### DISCUSSION TITLE
Publish GrandOrgue on Flathub

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#394] [General]
Originally posted 2021-03-28 13:29:14.203000 by [Laércio de Sousa](https://github.com/lbssousa) in GrandOrgue discussion forum on SourceForge<br><br>Hello there!

I&#039;ve just made a recipe to package GrandOrgue as a Flatpak for local use in my Fedora Silverblue computer. I&#039;m planning to publish it on Flathub, but first I want to ask if you want to maintain it yourselves.

My recipe: https://github.com/lbssousa/br.nom.sousa.laercio.GrandOrgue

Instructions for publishing on Flathub: https://github.com/flathub/flathub/wiki/App-Submission

That said, current grandorgue.appdata.xml file may need to be updated. See these guidelines: https://github.com/flathub/flathub/wiki/AppData-Guidelines There&#039;s an updated copy in my recipe above (I&#039;ve just rewritten the deprecated entries, but still need to fill the entries for app screenshots).

**Comment by larspalo** [comment]
Originally posted 2021-04-16 16:21:28.779000 by [Laércio de Sousa](https://github.com/lbssousa) in GrandOrgue discussion forum on SourceForge<br><br>App submission to Flathub currently in progress:

https://github.com/flathub/flathub/pull/2260

I&#039;ve included two patches in the packaging process you may want to apply upstream:

https://github.com/lbssousa/flathub/tree/net.sourceforge.GrandOrgue/patches

**Reply by larspalo** [reply]
Originally posted 2021-04-25 09:38:11.115000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>Hi, Laercio!

Could you make a PR with your patches  to my fork of GrandOrgue on GitHub: https://github.com/oleg68/GrandOrgue ?

**Reply by larspalo** [reply]
Originally posted 2021-04-26 12:56:57.092000 by [Laércio de Sousa](https://github.com/lbssousa) in GrandOrgue discussion forum on SourceForge<br><br>Long time no see, Oleg! Are you the one who worked on non-KMS multi-seat patches for X.Org I helped to send upstream?

I&#039;ll open a PR to your repo today.

**Reply by larspalo** [reply]
Originally posted 2021-04-26 16:09:03.599000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>Yes. It was me. I use multi-seat especially with GrandOrgue (one normal desktop seat and one - an organ console: https://b0.icdn.ru/o/oleg68/4/70011774fFw.jpg)

**Reply by larspalo** [reply]
Originally posted 2021-04-27 10:55:25.330000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>This release is compatible with FlatPack: https://github.com/oleg68/GrandOrgue/releases/tag/0.3.1.2338-3.os

**Reply by Rijaspb** [reply]
visit us on https://www.rijasexports.com/

**Comment by larspalo** [comment]
Originally posted 2021-04-19 16:21:59.053000 by [Laércio de Sousa](https://github.com/lbssousa) in GrandOrgue discussion forum on SourceForge<br><br>Now published on Flathub:

https://flathub.org/apps/details/net.sourceforge.GrandOrgue

**Comment by larspalo** [comment]
Originally posted 2021-04-20 07:21:48.442000 by *anonymous in GrandOrgue discussion forum on SourceForge<br><br>Laércio, thank you for doing this! I made a quick test in two VMs (Ubuntu and Manjaro) and it worked with charm.

Who updates the SourceForge website? It would be helpful to add this link to the &quot;Files&quot; section.

**Comment by lbssousa** [comment]
I've just updated [GrandOrgue package at Flathub](https://github.com/flathub/net.sourceforge.GrandOrgue) to match latest upstream release here in GitHub. From this point, you may want to take the ownership of this package and manage its next releases.

There are some actions that can be taken on the upstream side to make Flathub package maintenance easier:

* Keep `<releases>` section at file `resources/GrandOrgue.appdata.xml` up-to-date (I'm providing my own file at https://github.com/flathub/net.sourceforge.GrandOrgue/blob/master/GrandOrgue.appdata.xml for now)
* Migrate screenshots from Sourceforge to GitHub, or provide new ones here. Flathub build process may fail if connection to sourceforge.net (to validate screenshots registered in `GrandOrgue.appdata.xml`) fails.

**Comment by oleg68** [comment]
> Migrate screenshots from Sourceforge to GitHub, or provide new ones here

Will be done with #24

**Comment by Rijaspb** [comment]
https://www.rijasexports.com/

**Reply by larspalo** [reply]
Spam?

**Reply by rousseldenis** [reply]
Seems to be. We can block users in organization settings/security
